apps/emqx_management/src/emqx_mgmt_api_trace.erl

```diff
@@ -450,32 +451,37 @@ update_trace(put, #{bindings := #{name := Name}}) ->
 %% if HTTP request headers include accept-encoding: gzip and file size > 300 bytes.
 %% cowboy_compress_h will auto encode gzip format.
 download_trace_log(get, #{bindings := #{name := Name}, query_string := Query}) ->
-    Nodes =
-        case parse_node(Query, undefined) of
-            {ok, undefined} -> mria_mnesia:running_nodes();
-            {ok, Node0} -> [Node0];
-            {error, not_found} -> mria_mnesia:running_nodes()
-        end,
-    case emqx_trace:get_trace_filename(Name) of
-        {ok, TraceLog} ->
-            TraceFiles = collect_trace_file(Nodes, TraceLog),
-            ZipDir = emqx_trace:zip_dir(),
-            Zips = group_trace_file(ZipDir, TraceLog, TraceFiles),
-            FileName = binary_to_list(Name) ++ ".zip",
-            ZipFileName = filename:join([ZipDir, FileName]),
-            {ok, ZipFile} = zip:zip(ZipFileName, Zips, [{cwd, ZipDir}]),
-            %% emqx_trace:delete_files_after_send(ZipFileName, Zips),
-            %% TODO use file replace file_binary.(delete file after send is not ready now).
-            {ok, Binary} = file:read_file(ZipFile),
-            ZipName = filename:basename(ZipFile),
-            _ = file:delete(ZipFile),
-            Headers = #{
-                <<"content-type">> => <<"application/x-zip">>,
-                <<"content-disposition">> => iolist_to_binary("attachment; filename=" ++ ZipName)
-            },
-            {200, Headers, {file_binary, ZipName, Binary}};
+    case parse_node(Query, []) of
+        {ok, Node} ->
+            case emqx_trace:get_trace_filename(Name) of
+                {ok, TraceLog} ->
+                    case collect_trace_file(Node, TraceLog) of
+                        {error, not_found} ->
+                            ?BAD_REQUEST('NODE_ERROR', <<"Node not found">>);
```

Comment:
is this really a "Node not found" error ?
the node is found, but failed to do RPC, should be a 500 instead?

Reply:
That's misleading, the node was not "found" - at that point we just know it is a known atom. But it could be sth like `cowboy_rest` and would still be treated like a node name.

Reply:
I just realised, in `parse_node` we could always check if `Node` is a member of `mria_mnesia:running_nodes()`. Would that make sense?